src/app/src/components/DrawMap/EditToolbar.js

```diff
@@ -51,9 +51,17 @@ export default function EditToolbar() {
                 borderColor='gray.50'
                 borderRadius='6px'
                 boxShadow='lg'
+                cursor='default'
             >
                 <Flex>
-                    {polygon ? (
+                    {addPolygonMode ? (
+                        <Button
+                            onClick={() => dispatch(cancelAddPolygon())}
+                            w={40}
+                        >
+                            Cancel
+                        </Button>
+                    ) : polygon ? (
                         <Button
                             onClick={editDialogController.open}
```

Comment:
On that note, let's add a min-width as well to further solidify and protect from the toolbar changing unless needed for very long named polygons:

```suggestion
                            minW={40}
                            onClick={editDialogController.open}
```

<img width="1120" alt="image" src="https://user-images.githubusercontent.com/1430060/186525024-bf137201-868a-4b77-93f4-a0e3d06e58c9.png">

Reply:
57e0cbe085af90d9b9b6b2aa49503028625cbeab

I extracted to a constant in case we want to change the value later.